Mathlib/Analysis/Convex/Cone/Proper.lean

```diff
@@ -69,62 +68,96 @@ theorem closure_eq {K L : ConvexCone 𝕜 E} : K.closure = L ↔ closure (K : Se
 
 end ConvexCone
 
-/-- A proper cone is a convex cone `K` that is nonempty and closed. Proper cones have the nice
-property that the dual of the dual of a proper cone is itself. This makes them useful for defining
-cone programs and proving duality theorems. -/
+namespace PointedCone
+
+variable {𝕜 : Type*} [OrderedSemiring 𝕜]
+
+variable {E : Type*} [AddCommMonoid E] [TopologicalSpace E] [ContinuousAdd E] [Module 𝕜 E]
+  [ContinuousConstSMul 𝕜 E]
+
+lemma closure_aux (K : PointedCone 𝕜 E) : (K : ConvexCone 𝕜 E).closure.Pointed :=
+  subset_closure $ PointedCone.toConvexCone_pointed _
+
+/-- The closure of a pointed cone inside a topological space as a pointed cone. This
+construction is mainly used for defining maps between proper cones. -/
+protected def closure (K : PointedCone 𝕜 E) : PointedCone 𝕜 E :=
+  ConvexCone.toPointedCone K.closure_aux
+
+@[simp, norm_cast]
+theorem coe_closure (K : PointedCone 𝕜 E) : (K.closure : Set E) = closure K :=
+  rfl
+
+@[simp]
+protected theorem mem_closure {K : PointedCone 𝕜 E} {a : E} :
+    a ∈ K.closure ↔ a ∈ closure (K : Set E) :=
+  Iff.rfl
+
+@[simp]
+theorem closure_eq {K L : PointedCone 𝕜 E} : K.closure = L ↔ closure (K : Set E) = L :=
+  SetLike.ext'_iff
+
+end PointedCone
+
+/-- A proper cone is a pointed cone `K` that is closed. Proper cones have the nice property that
+the dual of the dual of a proper cone is itself. This makes them useful for defining cone programs
+and proving duality theorems. -/
 structure ProperCone (𝕜 : Type*) (E : Type*) [OrderedSemiring 𝕜] [AddCommMonoid E]
-    [TopologicalSpace E] [SMul 𝕜 E] extends ConvexCone 𝕜 E where
-  nonempty' : (carrier : Set E).Nonempty
+    [TopologicalSpace E] [Module 𝕜 E] extends Submodule {c : 𝕜 // 0 ≤ c} E where
```

Comment:
I assume `extends PointedCone` is not allowed here? I think that might be a reasonable feature request to core...

Reply:
Yes, it throws an error saying `'PointedCone' is not a structure.`

Should I open an issue in the Lean repo? Or just post on Zulip?